ts/nni_manager/common/training_service_v3.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ *  TODO
+ *  This type is planned to be an interface.
+ *  To minimize the change, it is currently identical to TrialJobMetric.data
+ **/
+export type Metric = string;
+
+/**
+ *  TODO
+ *  This type is planned to be an interface.
+ *  To minimize the change, it is currently identical to HyperParameters.value
+ **/
+export type Parameter = string;
+
+export interface EnvironmentInfo {
+    id: string;
+    // TODO
+}
+
+export interface TrainingServiceV3 {
+    /**
+     *  Invoked during experiment initialization.
+     *
+     *  It should verify the config and raise error if the specified training resource is not available.
+     *
+     *  It should not start daemon on worker machine.
+     *  If another component of the experiment failed to initialize,
+     *  the process will abort without invoking any clean up function.
```

Comment:
this implies that there should be clean-up but you decide not to do the clean-up, why?

Reply:
This method is invoked during initializing stage.
Failing to initialize means NNI manager is in an unexpected condition, so nothing is guaranteed to work, and halting is the safest action.